feat: make.com/integromat app

aar2dee2 opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes #1550 
/attempt #1550

<!-- Please provide a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

**Environment**: Staging(main branch) / Production

## Type of change

<!-- Please delete bullets that are not relevant. -->


- New feature (non-breaking change which adds functionality)

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- [ ] Test A
- [ ] Test B

## Checklist

<!-- Please remove all the irrelevant bullets to your PR -->

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

aar2dee2 committed:
start make app integration

joeauyeung commented:
Hi @aar2dee2, great start on the app so far. I'm going to convert this PR to a draft as this isn't fully flushed out yet. Once you are done feel free to mark this PR as ready for review.

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 commented:
> @aar2dee2 whats missing here to be ready for review?

Hi @PeerRich 
Still working on this, I'll push some commits in a few hours.

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
setup integration

aar2dee2 committed:
add relevant env vars

aar2dee2 committed:
update app metadata

aar2dee2 committed:
import setup route in app setups

aar2dee2 committed:
fix typo

aar2dee2 committed:
add app store imports

aar2dee2 committed:
fix module import error

aar2dee2 commented:
There a number of handlers that import the nodeScheduler from the `Zapier` app to perform an action when bookings are created/deleted etc. Do we want to follow the same approach for the `Make` integration?
- [`requestReschedule`](https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/bookings/requestReschedule.handler.ts)
- [`handleNewBooking`](https://github.com/calcom/cal.com/blob/main/packages/features/bookings/lib/handleNewBooking.ts)
- [`handleConfirmation`](https://github.com/calcom/cal.com/blob/main/packages/features/bookings/lib/handleConfirmation.ts)
- [`handleCancelBooking`](https://github.com/calcom/cal.com/blob/main/packages/features/bookings/lib/handleCancelBooking.ts)
- [`sendPayload`](https://github.com/calcom/cal.com/blob/main/packages/features/webhooks/lib/sendPayload.ts)


Will we need to handle `Make` as `Zapier` is handled in the files below?
- [`delete.handler`](https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/apiKeys/delete.handler.ts)
- [`deleteCredential.handler`](https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts)
- [`list.handler`](https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/apiKeys/list.handler.ts)


@PeerRich tagging you, not sure who's the right person to answer this.

aar2dee2 commented:
Working integration with Google Sheets:
https://www.loom.com/share/dd74c25570c2438a8d0e2fc354df73af

aar2dee2 committed:
update make readme

aar2dee2 committed:
Merge branch 'main' into integromat-app

PeerRich commented:
> tagging you, not sure who's the right person to answer this.

@CarinaWolli built the zapier app and can help

CarinaWolli commented on packages/app-store/make/api/subscriptions/listBookings.ts:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  try {
+    const bookings = await prisma.booking.findMany({
```
We should make a reusable logic for that and import it for Zapier and here. Otherwise, if we make changes in the fields returned for the bookings we would always need to make those changes in both places.

aar2dee2 replied:
Okay, so move the module to a different location and accept the app name/slug as a function param?

Which dir would you move the module to?

aar2dee2 replied:
also, this can be done in this PR itself or should I open another one?

CarinaWolli replied:
I think we can put the reusable code inside the `_utils` folder in `packages/app-store`. And we should have that booking query from `/listBookings` already covered in this PR, so we can assure it will always return the same for Zapier and Make. 

If the nodeScheduler is just a copy of the zapier one, then let's also make sure to make it reusable and not just copy it

aar2dee2 replied:
thanks! I'll make changes as per this.

CarinaWolli commented on packages/app-store/make/lib/nodeScheduler.ts:
```diff
@@ -0,0 +1,94 @@
+
+import prisma from "@calcom/prisma";
+
+export async function scheduleTrigger(
```
This looks like the exact same we have for zapier. We should definitely reuse that as well

CarinaWolli reviewed (commented):
This looks like a lot of copy/paste from the Zapier code. Can you try to minimize the duplicate code and create some reusable functions instead? 

And to answer your question: 
We should make sure that the logic we already have in these files to trigger the webhoks with `sendPayload` will also work for the Make app

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
move scheduler to app-store utils

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
move add subscription to node scheduler

aar2dee2 commented:
update: running into an issue in setting up the webhooks in the Make integration. Opened a discussion in the Make community: https://community.make.com/t/setting-up-webhook-instant-trigger-for-a-custom-app/11812

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
move delete subscription to scheduler

aar2dee2 commented:
Working integrations:

1. [List Bookings](https://www.loom.com/share/5a7f93bd2aa749f3afb6e6d1496eb62f?sid=43c3d566-d4bf-460e-89a0-20c3968f9bc0)
2. [Booking Created](https://www.loom.com/share/8515d3f70b0945058331995d6a2ced84?sid=cbdfe7dc-a0db-4819-a1b9-dfa271b90186)
3. [Booking Rescheduled](https://www.loom.com/share/1d50699e6dde44a58219836a7b540522?sid=4c98fae6-2b5b-43ab-9639-c87781715119)
4. [Booking Cancelled](https://www.loom.com/share/f8c422b84b3b4a0f862e357dae38ee0d?sid=94e2ff74-6251-4296-b7f4-d8c7b7156d75)
5. [Delete Subscription](https://www.loom.com/share/ebb9073713564fe2bdc662417aa08f4f?sid=7ddaf0b5-ff52-479d-a45b-a9f1fa29d8c1)

aar2dee2 committed:
subscribe unsubscribe in zapier using common nodeScheduler

aar2dee2 commented:
I'll clean up the `README` once the code is reviewed.
Link to published app on Make.com - https://www.make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510

aar2dee2 committed:
Merge branch 'main' into integromat-app

PeerRich commented:
> I'll clean up the `README` once the code is reviewed. Link to published app on Make.com - [make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510](https://www.make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510)

amazing! We're very excited about this cc @CarinaWolli

PeerRich committed:
Merge branch 'main' into integromat-app

CarinaWolli commented:
@aar2dee2 can you take a look at the lint errors?

CarinaWolli committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
fix lint errors

aar2dee2 commented:
> @aar2dee2 can you take a look at the lint errors?

done!

shivamklr commented on packages/app-store/make/static/1.jpeg:
Kindly add relevant static images. You can add the screenshot of setting up the process.

shivamklr replied:
Completed
- [x]  Tested GET List Bookings with generated API key. 
Pending
- [ ] Test Webhook trigger on Created, Cancelled, Rescheduled.
- [ ] Test how it will work if both apps, zapier and make, are enabled.

PeerRich committed:
Merge branch 'main' into integromat-app

PeerRich commented on .vscode/settings.json:
```diff
@@ -7,5 +7,22 @@
   "spellright.language": ["en"],
   "spellright.documentTypes": ["markdown", "typescript", "typescriptreact"],
-  "tailwindCSS.experimental.classRegex": [["cva\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]]
+  "tailwindCSS.experimental.classRegex": [["cva\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]],
+  "apps-sdk.login": true,
+  "apps-sdk.environments": [
+    {
+      "uuid": "3409f038-ddc7-45dc-8b67-0419987169c5",
+      "url": "us1.make.com/api",
+      "name": "NewPath Sandbox",
+      "apikey": "75fa5a40-beff-494d-8904-88bf09c34cc9",
+      "version": 2
+    }
+  ],
+  "apps-sdk.environment": "3409f038-ddc7-45dc-8b67-0419987169c5",
+  "json.schemas": [],
+  "githubPullRequests.pullBranch": "never",
+  "git.enableSmartCommit": true,
+  "workbench.startupEditor": "none",
+  "apps-sdk.apikey": "75fa5a40-beff-494d-8904-88bf09c34cc9",
+  "http.systemCertificates": false
```
what is all of this?

aar2dee2 replied:
reverted 

(was setup for the Make VSCode integration)

aar2dee2 committed:
revert settings.json

aar2dee2 committed:
update icon

aar2dee2 committed:
add app screenshots

aar2dee2 committed:
fix app description

shivamklr reviewed (commented):
Kindly resolve the conflicts with the main branch.

PeerRich commented:
/bonus 20 hey @aar2dee2 would you mind fixing the merge conflicts?

aar2dee2 commented:
> /bonus 20 hey @aar2dee2 would you mind fixing the merge conflicts?

Hey @PeerRich
This is on my to-dos for the weekend!

aar2dee2 commented:
Hi, I'd love some help on a `yarn` error I'm running into.

On running `git commit` I get:
```sh
Usage Error: Couldn't find the node_modules state file - running an install might help (findPackageLocation)

$ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] <scriptName> ...
husky - pre-commit hook exited with code 1 (error)
```

`yarn install` fails with errors
<img width="1277" alt="Screenshot 2023-08-03 at 5 34 16 PM" src="https://github.com/calcom/cal.com/assets/85004512/e2e24387-c164-4194-a917-763e98625e73">

my yarn version is `3.6.1` and node version `v18.15.0`

I checked that `.yarnrc.yml` is as per the Cal.com repo.

shivamklr commented:
@aar2dee2 I recommend doing two things, 

1. Refrain from pushing `yarn.lock` in the PR. 
2. Use `yarn clean` and then `yarn` again to install deps again.

aar2dee2 commented:
> @aar2dee2 I recommend doing two things,
> 
> 1. Refrain from pushing `yarn.lock` in the PR.
> 2. Use `yarn clean` and then `yarn` again to install deps again.

Thanks for getting back @shivamklr 

I deleted `yarn.lock`, ran `yarn install` again. Now `git commit` fails with the following error
<img width="1343" alt="Screenshot 2023-08-03 at 5 51 33 PM" src="https://github.com/calcom/cal.com/assets/85004512/8202c9f5-3f34-459a-8d6a-685b2ebd8f9e">

aar2dee2 committed:
merge main

aar2dee2 commented:
I've pulled the latest changes, made modifications to the `nodeScheduler` file and updated the `zapier` and `make` code.
The make integration is running with the new edits.

I was still getting `eslint` errors when the pre-commit runs. `yarn lint` runs fine, but the pre-commit hook fails. This commit was made using the `--no-verify` flag. Here are screenshots of the errors:
<img width="739" alt="Screenshot 2023-08-14 at 11 20 31 AM" src="https://github.com/calcom/cal.com/assets/85004512/6ae13279-eaa0-4ea0-ac07-bea6940b84da">

<img width="1016" alt="Screenshot 2023-08-14 at 11 20 17 AM" src="https://github.com/calcom/cal.com/assets/85004512/02ada62d-f3ad-4926-8922-a2e7e4144265">

<img width="1131" alt="Screenshot 2023-08-14 at 11 20 09 AM" src="https://github.com/calcom/cal.com/assets/85004512/3ec17614-bdd9-41bf-a807-a09036bc1972">

<img width="1001" alt="Screenshot 2023-08-14 at 11 20 03 AM" src="https://github.com/calcom/cal.com/assets/85004512/4f89adba-a4e7-42a9-95e8-f56cbbe9545e">

aar2dee2 committed:
fix type errors

aar2dee2 committed:
update app code

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 commented:
Hey Cal team,
would love to complete and close this soon

CarinaWolli commented:
How can I test that? And that would also mean we need to list the cal.com app on make.com, currently, it's just a private app in your make account, right?

aar2dee2 commented:
> @aar2dee2 How can I test that? And that would also mean we need to list the cal.com app on make.com, currently, it's just a private app in your make account, right?

Hi @CarinaWolli 

I published the app on Make, so you should be able to see it if you search for `Cal.com`. Here's a public link if required:
https://www.make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510

I tested the app with a Google Sheets integration for all the events. Videos should be there earlier in the comments on this PR.

CarinaWolli committed:
Delete .gitkeep

CarinaWolli commented on packages/app-store/_utils/nodeScheduler.ts:
```diff
@@ -0,0 +1,266 @@
+  const createSubscription = await prisma.webhook.create({
+    data: {
+      id: v4(),
+      userId: appApiKey.userId,
```
we are missing teamId here

CarinaWolli commented on packages/app-store/make/api/subscriptions/deleteSubscription.ts:
```diff
@@ -0,0 +1,39 @@
+  id: z.string(),
+});
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
```
when is deleteSubscribtion called in make? I don't ever see the webhook deleted in the database

aar2dee2 replied:
In the sidebar, there's a section called `Webhooks`. You need to select the webhook from there and delete each individually. I've added a line about this in the readme.

CarinaWolli reviewed (changes requested):
Thank you for this PR, finally got the chance to fully test and review it. Overall it's looking pretty good, I left you some comments and questions. Some of them are blocking so requesting changes 🙏

aar2dee2 committed:
delete unused template files

aar2dee2 committed:
get app invite link from env vars

aar2dee2 committed:
chore: handle error, cleanup readme, address review comments

aar2dee2 committed:
fix: update link in readme

aar2dee2 committed:
revert yarn.lock

aar2dee2 committed:
Merge branch 'main' into integromat-app

aar2dee2 committed:
fix type errors

aar2dee2 commented:
> Thank you for this PR, finally got the chance to fully test and review it. Overall it's looking pretty good, I left you some comments and questions. Some of them are blocking so requesting changes 🙏

Hey @CarinaWolli 
I've addressed all comments, lmk if any other updates are needed.

Thanks for the review!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=8897) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=8897) before we can accept your contribution.<br/>**0** out of **3** committers have signed the CLA.<br/><br/>:x: PeerRich<br/>:x: CarinaWolli<br/>:x: aar2dee2<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=8897) it.</sub>

PeerRich commented:
/bonus 20 @aar2dee2 can you fix the merge conflicts?

PeerRich commented:
after addressed merge conflict we can merge this

aar2dee2 committed:
Merge branch 'main' into integromat-app

PeerRich committed:
Merge branch 'main' into integromat-app

zomars commented on packages/app-store/_utils/nodeScheduler.ts:
```diff
@@ -0,0 +1,298 @@
+import type { Prisma } from "@prisma/client";
+import schedule from "node-schedule";
```
Just a sidenote. We're planning to migrate away from "node-schedule" aren't we? Not a blocker but something to have in mind as a follow up. cc @alannnc @CarinaWolli

CarinaWolli replied:
There is already a PR up for replacing node-schedule: https://github.com/calcom/cal.com/pull/10946

I will review that today and can help resolving conflicts

zomars commented on .env.appStore.example:
```diff
@@ -108,6 +108,11 @@ VITAL_REGION="us"
 
+#  - MAKE
+# Used for the Make integration
+# @see https://github.com/calcom/cal.com/blob/main/packages/app-store/make/README.md
```
Didn't see any instructions on how to get the actual MAKE_INVITE_LINK

CarinaWolli replied:
here is the invite link to test: [make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510](https://www.make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510)

We need to add invite link in general to the setup page as long this app is not yet public, left a comment here: https://github.com/calcom/cal.com/pull/8897#discussion_r1304787291

hariombalhara replied:
We don't need to add env variables for apps now. So removing it. We do need the instruction on where to get the invite link though.

aar2dee2 replied:
We're using the invite link in Carina's comment above. I'll open another PR to push the app for review on Make so we have a public app. Until then we can use this invite link: [make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510](https://www.make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510)

Pls lmk if any edits are needed

hariombalhara replied:
Yeah. We just need to add that link at admin/apps locally.
<img width="1310" alt="Screenshot 2023-08-31 at 12 42 33 PM" src="https://github.com/calcom/cal.com/assets/1780212/cf05b7db-e66f-4d4c-b35b-4baa9d199f36">

zomars reviewed (dismissed):
Code wise LGTM but couldn't make it work locally as I couldn't find info on how to set the MAKE_INVITE_LINK. The README states instructions from the USER side but not for the SysAdmin to setup this app.

CarinaWolli commented on packages/app-store/make/pages/setup/index.tsx:
```diff
@@ -0,0 +1,177 @@
+                      {t("make_invite_link")}
+                    </a>
+                  </li>
+                )}
```
@aar2dee2  That's still open, we need to add the invite link otherwise cal.com app won't show up in make.com

aar2dee2 replied:
Hey @CarinaWolli 
I'm using the invite link from env vars here
https://github.com/calcom/cal.com/pull/8897/files#diff-353f499de80623fd657b800f303fa8b15f55de5e2818cb07b3cfacc556a1e609R12
and then accessing it here:
https://github.com/calcom/cal.com/pull/8897/files#diff-353f499de80623fd657b800f303fa8b15f55de5e2818cb07b3cfacc556a1e609R116

do we need to implement this another way?

CarinaWolli replied:
Currently, we only show the invite link for our dev environment (that's was needed for zapier). But for make we need to always show the invite links as this app is not yet publicly available

So removing `isCalDev &&` should do it

aar2dee2 replied:
fixed this.

CarinaWolli reviewed (changes requested):
After invite link is added to setup page, I can approve 🙏

PeerRich commented:
> After invite link is added to setup page, I can approve 🙏

can you do it @CarinaWolli ?

PeerRich committed:
Merge branch 'main' into integromat-app

hariombalhara commented on packages/prisma/seed-app-store.ts:
```diff
@@ -337,7 +337,11 @@ export default async function main() {
     });
   }
-
+  if (process.env.MAKE_INVITE_LINK) {
+    await createApp("make", "make", ["automation"], "make_automation", {
+      invite_link: process.env.MAKE_INVITE_LINK,
+    });
+  }
```
We don't need to do this. Infact, we don't want to do this. Whether an app is enabled(and seeded) or not is controlled through the admin/apps interface.

hariombalhara committed:
Update packages/prisma/seed-app-store.ts

hariombalhara committed:
Update .env.appStore.example

hariombalhara commented on .env.appStore.example:
```diff
@@ -108,6 +108,7 @@ VITAL_REGION="us"
 # @see https://github.com/calcom/cal.com/blob/main/packages/app-store/zapier/README.md
 ZAPIER_INVITE_LINK=""
 
+
```
```suggestion
```

hariombalhara committed:
Update .env.appStore.example

hariombalhara commented on packages/app-store/make/README.md:
```diff
@@ -0,0 +1,8 @@
+# Setting up Make Integration
```
@aar2dee2 I don't see the instructions on how to create the app on make.com.
We have something similar for zapier https://github.com/aar2dee2/cal.com/blob/6bfabe41e5073ff3a9e38d577ef6d72be2d6eca8/packages/app-store/zapier/README.md#L8

aar2dee2 replied:
I  just updated the readme. Request @CarinaWolli to look at point 5 - how did add the app to scenarios once you had the invite link?
[`5. Search for `Cal.com` in the apps list and select from the list of triggers - Booking Created, Booking Deleted, Booking Rescheduled, Meeting Ended`](https://github.com/calcom/cal.com/pull/8897/files#diff-01baf41f650a10552d5968a62b68c6d64b919b57c792ce7b9b4a55c36211835aR8)

hariombalhara commented on packages/app-store/_utils/nodeScheduler.ts:
```diff
@@ -0,0 +1,298 @@
+export async function deleteSubscription({
+  appApiKey,
+  webhookId,
+  appName,
```
nit: We should use same name for the same entity. `addSubscription` calls it `appId`. I like that name better.

aar2dee2 committed:
update app readme

aar2dee2 committed:
fix param name in deleteSubcription

aar2dee2 committed:
fix listBookings handler

aar2dee2 commented on packages/app-store/_utils/nodeScheduler.ts:
```diff
@@ -0,0 +1,298 @@
+}
+export async function listBookings(appApiKey: ApiKey) {
+  try {
+    const where: Prisma.BookingWhereInput = {};
```
fixed

hariombalhara commented on packages/app-store/make/api/add.ts:
```diff
@@ -0,0 +1,39 @@
+  }
+  const appType = "make_automation";
+  try {
+    const alreadyInstalled = await prisma.credential.findFirst({
+      where: {
+        type: appType,
+        userId: req.session.user.id,
+      },
+    });
+    if (alreadyInstalled) {
+      throw new Error("Already installed");
+    }
+    const installation = await prisma.credential.create({
+      data: {
+        type: appType,
+        key: {},
+        userId: req.session.user.id,
+        appId: "make",
+      },
+    });
+    if (!installation) {
```
I don't think we are doing any OAuth in there. We are just creating credentials as a means to mark the app installed. For that we have AppDeclarativeHandler. See [this](https://github.com/aar2dee2/cal.com/blob/6bfabe41e5073ff3a9e38d577ef6d72be2d6eca8/packages/app-store/skiff/api/add.ts#L21) as an example

hariombalhara replied:
That's already checked [here] when `createDefaultInstallation` is used (https://github.com/aar2dee2/cal.com/blob/6bfabe41e5073ff3a9e38d577ef6d72be2d6eca8/apps/web/pages/api/integrations/%5B...args%5D.ts#L12)

hariombalhara replied:
The job of `createDefaultInstallation` is to just create the installation. Additional checks are done `defaultIntegrationAddHandler`

aar2dee2 replied:
thanks!

edited this.

hariombalhara commented on packages/app-store/make/pages/setup/index.tsx:
```diff
@@ -0,0 +1,174 @@
+const MAKE = "make";
+const appInviteLink = process.env.MAKE_INVITE_LINK; //TODO Add MAKE_INVITE_LINK to env vars
+
+export default function MakeSetup() {
```
nit: This is almost identical with Zapier. I think it would make sense to have a common component used by both
![image](https://github.com/calcom/cal.com/assets/1780212/892d0c0c-5377-4f33-a22f-b42940267481)

We can have something like `ApiKeysSetup` that would take care of the following UI
![image](https://github.com/calcom/cal.com/assets/1780212/a63eddcd-be9d-42b8-a782-62c352859440)

The instructions can be kept per app

aar2dee2 replied:
where would you add this component?

hariombalhara replied:
I guess app-store/_components would be the right place for now as I don't see any usage as of now across the webapp.

aar2dee2 replied:
will open another PR for this

PeerRich commented:
there is a merge conflcit with nodeScheduler

```
<<<<<<< integromat-app -- Incoming Change
import { cancelScheduledJobs } from "@calcom/app-store/_utils/nodeScheduler";
=======
>>>>>>> main -- Current Change
```


do we remove this now?

hariombalhara commented on packages/app-store/make/static/1.jpeg:
This and the next screenshots are old. They are just showing one API key

aar2dee2 replied:
<img width="1288" alt="Screenshot 2023-08-31 at 4 15 44 PM" src="https://github.com/calcom/cal.com/assets/85004512/0cfcc8fb-c45a-4d3a-b264-b7a88cc00f5d">

This is still the same in my setup. is there an example of the updated version?

aar2dee2 replied:
great!

hariombalhara commented on packages/features/bookings/lib/handleCancelBooking.ts:
```diff
@@ -3,9 +3,9 @@ import type { NextApiRequest } from "next";
 
 import appStore from "@calcom/app-store";
 import { getCalendar } from "@calcom/app-store/_utils/getCalendar";
+import { cancelScheduledJobs } from "@calcom/app-store/_utils/nodeScheduler";
```
Nice, I don't see dependency on a single app now. There is an abstraction now.

hariombalhara commented on turbo.json:
```diff
@@ -220,6 +220,7 @@
     "LARK_OPEN_APP_ID",
     "LARK_OPEN_APP_SECRET",
     "LARK_OPEN_VERIFICATION_TOKEN",
+    "MAKE_INVITE_LINK",
```
```suggestion
```

hariombalhara committed:
Update turbo.json

hariombalhara commented on packages/app-store/_utils/nodeScheduler.ts:
```diff
@@ -0,0 +1,296 @@
+
+    return updatedBookings;
+  } catch (err) {
+    return console.error(
```
nit: I think it might be old code but it's better to use logger. We can go for `logger.error` here

aar2dee2 replied:
made edits. pls check child logger label.

aar2dee2 committed:
use default installation handler to install app

hariombalhara commented on packages/app-store/make/pages/setup/index.tsx:
```diff
@@ -0,0 +1,174 @@
+import { Clipboard } from "@calcom/ui/components/icon";
+
+const MAKE = "make";
+const appInviteLink = process.env.MAKE_INVITE_LINK; //TODO Add MAKE_INVITE_LINK to env vars
```
We would need to get it from `appKeys` like [this](https://github.com/aar2dee2/cal.com/blob/6bfabe41e5073ff3a9e38d577ef6d72be2d6eca8/packages/app-store/jitsivideo/lib/VideoApiAdapter.ts#L16) as we are not setting in env now.

hariombalhara replied:
Sorted this one.

aar2dee2 replied:
seems like you fixed this?

hariombalhara replied:
Yeah.

hariombalhara commented:
@aar2dee2 FYI, nits aren't blocking. So, feel free to address them as per your convenience. A followup PR would also work.

aar2dee2 committed:
use logger for console logs

hariombalhara committed:
Fix inviteLink reading

hariombalhara committed:
Merge remote-tracking branch 'aar2dee2/integromat-app' into integromat-app

hariombalhara commented on packages/app-store/make/static/1.jpeg:
This screenshot is outdated. Doesn't reflect multiple API keys.

I feel like I left the comment earlier. But I can't see it 🤔

aar2dee2 commented:
> @aar2dee2 FYI, nits aren't blocking. So, feel free to address them as per your convenience. A followup PR would also work.

thanks!

aar2dee2 committed:
fix app setup handler

hariombalhara committed:
Fix type issue

hariombalhara reviewed (dismissed):
Code looks good to me now!! Great work @aar2dee2. Conflicts need to be resolved though before merge which I think @CarinaWolli will help with that.

aar2dee2 committed:
always show app invite link

Commit:
Merge branch 'main' into integromat-app

Commit:
fix type error

CarinaWolli committed:
Merge branch 'main' into integromat-app

Commit:
add make invite link

Commit:
Merge branch 'integromat-app' of https://github.com/aar2dee2/cal.com into integromat-app

CarinaWolli commented on packages/prisma/seed-app-store.ts:
```diff
@@ -337,7 +337,9 @@ export default async function main() {
   }
-
+  await createApp("make", "make", ["automation"], "make_automation", {
+    invite_link: "https://make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510",
```
i'll take a look at how we can get the app published, so we won't need the link no more

aar2dee2 replied:
There's a section to apply for review within the app I created on Make. I'll get it started and open another PR here so we can track the progress.
<img width="1426" alt="Screenshot 2023-09-01 at 9 15 15 AM" src="https://github.com/calcom/cal.com/assets/85004512/ba96df38-ff59-4a88-8031-2370bc183fb3">

CarinaWolli reviewed (approved):
I fixed the conflicts and made some small changes. Looks good to me now, great work 🙌🏻

CarinaWolli committed:
Merge branch 'main' into integromat-app

elie222 commented on apps/web/public/static/locales/en/common.json:
```diff
@@ -1089,6 +1092,7 @@
   "or_lowercase": "or",
   "nevermind": "Nevermind",
   "go_to": "Go to: ",
+  "invite_link_make": "<li>Go to<a>Make Invite Link</a> and install Cal.com appp</li>",
```
Typo here. It's app

PeerRich replied:
thx. will fix in a different PR

elie222 reviewed (commented):
Typo

PeerRich commented:
![image](https://github.com/calcom/cal.com/assets/8019099/0391ca37-21f0-4360-ab2e-6c5029ff4126)
 what do we need here @aar2dee2 ?

aar2dee2 commented:
> ![image](https://user-images.githubusercontent.com/8019099/265824663-0391ca37-21f0-4360-ab2e-6c5029ff4126.png) what do we need here @aar2dee2 ?

We need to copy the invite link from the app store README. Here's the link `https://www.make.com/en/hq/app-invitation/6cb2772b61966508dd8f414ba3b44510`


I'm thinking we can also add the invite link to the app description in the app store until we have a public app on Make?
<img width="523" alt="Screenshot 2023-09-06 at 9 59 08 AM" src="https://github.com/calcom/cal.com/assets/85004512/60c9ac4d-1ddd-4492-9a15-742d83a5c871">

CarinaWolli commented:
@aar2dee2  I think it should be fine to have the invite link on the Make setup page until the app is public. I somehow messed that up in this PR, but just opened a fix: https://github.com/calcom/cal.com/pull/11177

<img width="633" alt="Screenshot 2023-09-06 at 14 06 29" src="https://github.com/calcom/cal.com/assets/30310907/d0a3cab1-fe77-4a58-8acd-6932e02a054d">
